recipes/oatpp-libressl/all/conanfile.py

```diff
@@ -0,0 +1,90 @@
+from conans import ConanFile, CMake, tools
+from conans.errors import ConanInvalidConfiguration
+import os
+
+class OatpplibresslConan(ConanFile):
+    name = "oatpp-libressl"
+    license = "Apache-2.0"
+    homepage = "https://github.com/oatpp/oatpp-libressl"
+    url = "https://github.com/conan-io/conan-center-index"
+    description = "oat++ libressl library"
+    topics = ("conan", "oat++", "oatpp", "libressl")
+    settings = "os", "compiler", "build_type", "arch"
+    options = {"shared": [True, False], "fPIC": [True, False]}
+    default_options = {"shared": False, "fPIC": True}
+    generators = "cmake", "cmake_find_package"
+    exports_sources = "CMakeLists.txt"
+
+    _cmake = None
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    @property
+    def _build_subfolder(self):
+        return "build_subfolder"
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        if self.options.shared:
+            del self.options.fPIC
+        if self.settings.compiler.cppstd:
+            tools.check_min_cppstd(self, 11)
+
+        if self.settings.os == "Windows" and self.options.shared:
+            raise ConanInvalidConfiguration("oatpp-libressl can not be built as shared library on Windows")
+
+        if self.settings.compiler == "gcc" and tools.Version(self.settings.compiler.version) < "5":
+            raise ConanInvalidConfiguration("oatpp-libressl requires GCC >=5")
+
+    def requirements(self):
+        self.requires("oatpp/1.2.0")
+        self.requires("libressl/3.2.0")
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        os.rename("oatpp-libressl-{0}".format(self.version), self._source_subfolder)
+
+    def _configure_cmake(self):
+        if self._cmake:
+            return self._cmake
+
+        self._cmake = CMake(self)
+        self._cmake.definitions["OATPP_BUILD_TESTS"] = False
+        self._cmake.definitions["OATPP_MODULES_LOCATION"] = "CUSTOM"
+        self._cmake.definitions["OATPP_DIR_LIB"] = os.path.join(
+            self.deps_cpp_info["oatpp"].rootpath, self.deps_cpp_info["oatpp"].libdirs[0]
+        )
+        self._cmake.definitions["OATPP_DIR_SRC"] = self.deps_cpp_info["oatpp"].include_paths[0]
```

Comment:
Looking at the CMake https://github.com/oatpp/oatpp-libressl/blob/cd2e9a515131e5e7dc043c591e952e12cd63db2c/CMakeLists.txt#L35

🤔 Could we not use the "Installed" option with find_package generators being enabled?

Reply:
Yes, we could but it is not necessary. Look at this approved reciepe [opatpp-websocket](https://github.com/conan-io/conan-center-index/blob/15276f9a7a3add397b2e172538046ed6479684c5/recipes/oatpp-websocket/all/conanfile.py#L60) I was inspiring this recipe. With `CUSTOM` we manually set location of oatpp main module, With `INSTALLED` cmake does the same but with find()

Reply:
Ahhh, it might be worth a test so we could clean the other one too